test/ModifyLiquidity.t.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {Test} from "forge-std/Test.sol";
+import {Deployers} from "./utils/Deployers.sol";
+import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
+import {PoolKey} from "src/types/PoolKey.sol";
+import {IPoolManager} from "src/interfaces/IPoolManager.sol";
+import {IHooks} from "src/interfaces/IHooks.sol";
+import {Position} from "src/libraries/Position.sol";
+import {PoolId} from "src/types/PoolId.sol";
+
+contract ModifyLiquidityTest is Test, Deployers, GasSnapshot {
+    PoolKey simpleKey; // vanilla pool key
+    PoolId simplePoolId; // id for vanilla pool key
+
+    bytes32 SALT = hex"CAFF";
+
+    IPoolManager.ModifyLiquidityParams public LIQ_PARAM_NO_SALT =
+        IPoolManager.ModifyLiquidityParams({tickLower: -120, tickUpper: 120, liquidityDelta: 1e18, salt: 0});
+
+    IPoolManager.ModifyLiquidityParams public LIQ_PARAM_SALT =
+        IPoolManager.ModifyLiquidityParams({tickLower: -120, tickUpper: 120, liquidityDelta: 1e18, salt: SALT});
+
+    function setUp() public {
+        deployFreshManagerAndRouters();
+        deployMintAndApprove2Currencies();
+        (simpleKey, simplePoolId) = initPool(currency0, currency1, IHooks(address(0)), 3000, SQRT_RATIO_1_1, ZERO_BYTES);
+    }
+
+    function test_modifyLiquidity_samePosition_zeroSalt_isUpdated() public {
+        Position.Info memory position = manager.getPosition(
+            simplePoolId, address(modifyLiquidityRouter), LIQ_PARAM_NO_SALT.tickLower, LIQ_PARAM_NO_SALT.tickUpper, 0
+        );
+        assertEq(position.liquidity, 0);
+        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_NO_SALT, ZERO_BYTES, false, false);
```

Comment:
to make the tests cleaner to read, you can use the overloaded `.modifyLiquidity` which defaults the settle/take with ERC20 transfers

(would replace everywhere else in this test file too)

```suggestion
        modifyLiquidityRouter.modifyLiquidity(simpleKey, LIQ_PARAM_NO_SALT, ZERO_BYTES);
```